packages/ui-test-queries/src/utils/queries.ts

```diff
@@ -24,58 +24,78 @@
 
 import { firstOrNull } from './firstOrNull'
 import { getQueryResult } from './queryResult'
-import { parseQueryArguments } from './parseQueryArguments'
+import { parseQueryArguments, QueryArguments } from './parseQueryArguments'
 import {
   querySelectorAll,
   querySelectorFrames,
-  querySelectorParents
+  querySelectorParents,
+  SelectorOptions
 } from './selectors'
+import { BoundType } from './bindElementToUtilities'
 
-async function findWithLabel(...args) {
+export type QueryFunction<T> = (
+  element: Element | Document | DocumentFragment,
+  selector: string | undefined,
+  options: SelectorOptions
+) => T[]
+
+async function findWithLabel(...args: QueryArguments) {
   const { element, selector, options } = parseQueryArguments(...args)
   return find(element, `:withLabel("${selector}")`, options)
 }
 
-async function findWithText(...args) {
+async function findWithText(...args: QueryArguments) {
   const { element, selector, options } = parseQueryArguments(...args)
   return find(element, `:withText("${selector}")`, options)
 }
 
-async function findWithTitle(...args) {
+async function findWithTitle(...args: QueryArguments) {
   const { element, selector, options } = parseQueryArguments(...args)
   return find(element, `:withTitle("${selector}")`, options)
 }
 
-async function find(...args) {
-  return firstOrNull(await findAll(...args))
+async function find(...args: QueryArguments) {
+  // TODO remove "as any", but this will cause lots of TSC errors
+  // the main reason is that the function params are bound in bindElementToMethods
+  return firstOrNull(await findAll(...args)) as any
```

Comment:
Sadly adding this here breaks autocomplete totally. But removing the "any" cast from here would make type checking buggy, since methods in 'helpers' and 'queries' have a parameter (the element its added to) that is added automatically with function.bind().
I dont think we can solve this without large refactorings